### PR TITLE
Add a destroy function to the encoder

### PR DESCRIPTION
**Pull request by adambailey-**

### CODE REVIEW

File: sdks/js/src/classes/encoder.js [hunk 1]
```diff
@@ -50,6 +50,18 @@ class Encoder extends Emitter {
       buffers: data
     });
   }
+
+  destroy() {
+    this.encoderWorker.postMessage({
+      command: 'destroy'
```
Comment (vocoded) [inline sdks/js/src/classes/encoder.js:56]: destroy before close?

File: sdks/js/src/classes/encoder.js [hunk 2]
```diff
@@ -50,6 +50,18 @@ class Encoder extends Emitter {
...
+    });
+    this.encoderWorker.postMessage({
```
Comment (vocoded) [inline sdks/js/src/classes/encoder.js:58]: and are these synchronous posts?

File: sdks/js/src/classes/encoder.js (hunk 1 above)
Comment (adambailey-) [reply sdks/js/src/classes/encoder.js:56]: It's worded kinda funny...destroy doesn't destroy the worker, it frees all the opus memory. Close on the other hand closes the web worker

Comment (vocoded) [reply sdks/js/src/classes/encoder.js:56]: gotcha.  maybe a comment would help for future reference

File: sdks/js/src/classes/encoder.js (hunk 2 above)
Comment (adambailey-) [reply sdks/js/src/classes/encoder.js:58]: Good catch, no. We don't really need both `close` and `terminate`, I was just being careful. I'm gonna change the code slightly to only use `close`

**Pull request merged**